waiter/src/waiter/scheduler/kubernetes.clj

```diff
@@ -0,0 +1,605 @@
+            delete-result (delete-service this service)]
+        (swap! service-id->failed-instances-transient-store dissoc service-id)
+        (scheduler/remove-killed-instances-for-service! service-id)
+        delete-result)
+      (catch [:status 404] _
+        (log/warn "[delete-app] Service does not exist:" service-id)
+        {:result :no-such-service-exists
+         :message "Kubernetes reports service does not exist"})
+      (catch [:status 409] _
+        (log/warn "Kubernetes ReplicaSet conflict while deleting"
+                  {:service-id service-id})
+        {:result :conflict
+         :message "Kubernetes ReplicaSet conflict while deleting"})
+      (catch Throwable e
+        (log/warn "Kubernetes ReplicaSet conflict while deleting"
+                  {:service-id service-id})
+        {:result :error
+         :message "Internal error while deleting service"})))
+
+  (scale-app [this service-id scale-to-instances]
+    (ss/try+
+      (if-let [service (service-id->service this service-id)]
+        (do
+          (scale-service-up-to this service scale-to-instances)
+          {:success true
+           :status 200
+           :result :scaled
+           :message (str "Scaled to " scale-to-instances)})
+        (do
+          (log/error "Cannot scale missing service" service-id)
+          {:success false
+           :status 404
+           :result :no-such-service-exists
+           :message "Failed to scale missing service"}))
+      (catch [:status 409] _
+        {:success false
+         :status 409
+         :result :conflict
+         :message "Scaling failed due to repeated patch conflicts"})
+      (catch Throwable e
+        (log/error e "Error while scaling waiter service" service-id)
+        {:success false
+         :status 500
+         :result :failed
+         :message "Error while scaling waiter service"})))
+
+  (retrieve-directory-content [_ service-id instance-id _ relative-directory]
+    ;; TODO (#357) - get access to the working directory contents in the pod
+    [])
+
+  (service-id->state [_ service-id]
+    {:failed-instances (vals (get @service-id->failed-instances-transient-store service-id))
+     :killed-instances (scheduler/service-id->killed-instances service-id)})
+
+  (state [_]
+    {:service-id->failed-instances @service-id->failed-instances-transient-store}))
+
+(defn- start-auth-renewer
+  "Initialize the k8s-api-auth-str atom,
+   and optionally start a chime to periodically referesh the value."
+  [{:keys [refresh-delay-minutes refresh-fn]}]
+  (let [refresh (-> refresh-fn utils/resolve-symbol deref)
+        auth-update-fn (fn auth-update []
+                         (if-let [auth-str' (refresh)]
+                           (reset! k8s-api-auth-str auth-str')))]
+    (auth-update-fn)
+    (when [refresh-delay-minutes]
+      (assert (utils/pos-int? refresh-delay-minutes))
+      (-> refresh-delay-minutes
+          t/minutes
+          (du/start-timer-task auth-update-fn)))))
```

Comment:
We need to set the `delay-ms` option to trigger the timer task after the delay.

Reply:
Done.